app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -67,15 +67,31 @@ def reload_answer
   def publish_all
     graded_submissions = @assessment.submissions.with_graded_state
     if !graded_submissions.empty?
-      job =
-        Course::Assessment::Submission::PublishingJob.perform_later(@assessment, current_user).job
+      job = Course::Assessment::Submission::PublishingJob.
+            perform_later(@assessment, current_user).job
       redirect_to(job_path(job))
     else
       redirect_to course_assessment_submissions_path(current_course, @assessment),
                   notice: t('.notice')
     end
   end
 
+  # download either all of or a subset of submissions for an assessment.
```

Comment:
Would be good to add the specifications  on the params for `limit` or the new name (what does `my`, `student`, `others` do).